### PR TITLE
twitter oauth

### PR DESCRIPTION
**Pull request by preddy5**

### CODE REVIEW

File: bookie/models/social.py [hunk 1]
```diff
@@ -21,6 +22,20 @@ def get_all_connections(username):
...
+            username=username,
+            is_active=credentials['is_active'],
+            last_connection=credentials['last_connection'],
+            uid=credentials['uid'],
+            access_key=credentials['access_key'],
+            access_secret=credentials['access_secret'],
+            twitter_username=credentials['twitter_username'],
+            refresh_date=credentials['refresh_date'])
+        DBSession.add(tconnection)
+        return 'Successfully Done'
```
Comment (mitechie) [inline bookie/models/social.py]: this should return the tconnection object vs a message.

File: bookie/views/social.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+
+@view_config(route_name="twitter_connect",
+             renderer="/social/social_connect.mako")
+def twitter_connect(request):
+    """function to get users twitter oauth information"""
+
+    # Twitter Application details are imported from ini file
+    twitter_consumer_key = \
+        request.registry.settings.get('twitter_consumer_key', False)
```
Comment (mitechie) [inline bookie/views/social.py]: please shorten this by creating a variable for settings.

```
settings = request.registry.setting
twitter_consumer_key = settings.get('twitter_consumer_key', False)
```

**Comment by mitechie** [comment]
:+1: with the changes mentioned. Thanks for the hard work on this!

**Comment by sambuddhabasu** [comment]
Looks good to me :+1:

**Comment by preddy5** [comment]
:shipit:

**Pull request merged**